File: EXAMPLES/IDE/c#/banking-app.cs
using System;

class Program
{
    static void Main()
    {
        string name;
        string email;
        double initialBalance;

        Console.WriteLine("Welcome to SEAP-10 Banking App!");
        Console.WriteLine("1. Check Balance");
        Console.WriteLine("2. Deposit Money");
        Console.WriteLine("3. Withdraw Money");
        Console.WriteLine("4. Register New User");
        Console.WriteLine("5. Exit");

        Console.Write("Please enter your choice (1-5): ");
        string choice = Console.ReadLine();

        switch (choice)
        {
            case "1":
                Console.WriteLine($"Your current balance is {currentBalance}");
                break;
            case "2":
                Console.Write("Enter the amount to deposit: ");
                double depositAmount = Convert.ToDouble(Console.ReadLine());
                currentBalance += depositAmount;
                Console.WriteLine($"${depositAmount} deposited successfully.");
                // Update balance logic here
                break;
            case "3":
                Console.Write("Enter the amount to withdraw: ");
                double withdrawAmount = Convert.ToDouble(Console.ReadLine());
                if (withdrawAmount <= currentBalance)
                {
                    currentBalance -= withdrawAmount;
                    Console.WriteLine($"${withdrawAmount} withdrawn successfully.");
                }
                else
                {
                    Console.WriteLine("Insufficient funds.");
                }
                // Update balance logic here
                break;
            case "4":
                Console.Write("Enter your name: ");
                name = Console.ReadLine();
                Console.Write("Enter your email: ");
                email = Console.ReadLine();
                Console.Write("Enter initial balance: ");
                initialBalance = Convert.ToDouble(Console.ReadLine());
                // Add logic to register new user with the provided information
                break;
            case "5":
                Console.WriteLine("Thank you for using our banking app. Goodbye!");
                break;
            default:
                Console.WriteLine("Invalid choice. Please select a valid option.");
                break;
        }
    }
}

File: EXAMPLES/IDE/typscript/banking-app.js
import * as readlineSync from 'readline-sync';

// User registration information
let userName: string;
let userEmail: string;
let initialBalance: number;

// Function to clear the screen
function clearScreen(): void {
    console.clear();
}

// Initial balance for the new user
const openingBalance: number = 1000;
let currentBalance: number = openingBalance;
let previousBalance: number = 0;

// Displaying a selection menu for a banking app
console.log("Welcome to SEAP-10 Banking App!");
console.log("1. Check Balance");
console.log("2. Deposit Money");
console.log("3. Withdraw Money");
console.log("4. Exit");

// Get user input for selection
const choice: string = readlineSync.question("Please enter your choice (1-4): ");

// User's selection
if (choice === "1") {
    console.log(`Your current balance is ${currentBalance}`);
} else if (choice === "2") {
    const amount: number = parseFloat(readlineSync.question("Enter the amount to deposit: "));
    currentBalance += amount;
    console.log(`$${amount} deposited successfully.`);
    // Update balance logic here
} else if (choice === "3") {
    const amount: number = parseFloat(readlineSync.question("Enter the amount to withdraw: "));
    if (amount <= currentBalance) {
        currentBalance -= amount;
        console.log(`$${amount} withdrawn successfully.`);
    } else {
        console.log("Insufficient funds.");
    }
    // Update balance logic here
} else if (choice === "4") {
    console.log("Thank you for using our banking app. Goodbye!");
} else {
    console.log("Invalid choice. Please select a valid option.");
}